src/core/include/openvino/core/graph_util.hpp

```diff
@@ -226,9 +226,10 @@ std::vector<std::shared_ptr<Node>> topological_sort(T root_nodes) {
     std::unordered_set<Node*> nodes_done;
     std::vector<std::shared_ptr<Node>> result;
 
-    for (auto& node : root_nodes) {
-        nodes_to_do.push(node.get());
-    }
+    for (auto node = root_nodes.rbegin(); node != root_nodes.rend(); ++node ) {
```

Comment:
Why do you need the reverse order here?

Reply:
nodes_to_do is a std::stack. It reverse order of inputs and outputs